Modernize launcher bundling

getchoo opened the pull request:
Replaces fixup_bundle with Qt's deployment scripts and CMake's newer RUNTIME_DEPENDENCY_SET target, making it a bit easier to find and include linked dependencies with less code on our end

This also fixes some issues i've discovered, like:

- MSVC debug builds current shipping the release and debug version of each dll
- *All* Windows builds shipping OpenSSL still
- Linux install bundles not bundling everything they actually need
- Dependencies just not being found in some cases because of how few directories fixup_bundle searches currently

~~blocked **not** by #3872 due to issues with the cross version of windeployqt~~ Turns out it was actually an issue with CMake!

## Testing Checklist

- [x] Windows (MSVC)
- [x] Windows (MinGW)
- [ ] Windows (MSVC, cross compiled ARM64)?
- [x] Linux (Portable)
- [x] Linux (AppImage)
- [x] macOS

getchoo committed:
build: modernize launcher bundling

Replaces fixup_bundle with Qt's deployment scripts and CMake's newer
RUNTIME_DEPENDENCY_SET target, making it a bit easier to find and
include linked dependencies with less code on our end

Signed-off-by: seth <getchoo@tuta.io>

getchoo committed:
build: let cmake know when we're cross compiling on msvc

This (unsurprisingly) makes some things actually work as they're
expected to when cross compiling, like windeployqt

Signed-off-by: Seth Flynn <getchoo@tuta.io>

Trial97 commented:
linux portable fails:
![image](https://github.com/user-attachments/assets/58cbc7fb-5b96-4210-99de-b59427e0e74f)
-# the develop portable build works

Trial97 reviewed (changes requested):
linux portable fails:
```
Launcher Dir: /home/user/Downloads/PrismLauncher-portable
sort: /home/user/Downloads/PrismLauncher-portable/lib/libc.so.6: version `GLIBC_ABI_DT_RELR' not found (required by sort)
sort: /home/user/Downloads/PrismLauncher-portable/lib/libc.so.6: version `GLIBC_2.38' not found (required by sort)
grep: /home/user/Downloads/PrismLauncher-portable/lib/libc.so.6: version `GLIBC_ABI_DT_RELR' not found (required by grep)
grep: /home/user/Downloads/PrismLauncher-portable/lib/libc.so.6: version `GLIBC_2.38' not found (required by grep)
grep: /home/user/Downloads/PrismLauncher-portable/lib/libc.so.6: version `GLIBC_ABI_DT_RELR' not found (required by /usr/lib/libpcre2-8.so.0)
awk: /home/user/Downloads/PrismLauncher-portable/lib/libm.so.6: version `GLIBC_2.38' not found (required by awk)
awk: /home/user/Downloads/PrismLauncher-portable/lib/libc.so.6: version `GLIBC_ABI_DT_RELR' not found (required by awk)
awk: /home/user/Downloads/PrismLauncher-portable/lib/libc.so.6: version `GLIBC_2.38' not found (required by awk)
awk: /home/user/Downloads/PrismLauncher-portable/lib/libc.so.6: version `GLIBC_ABI_DT_RELR' not found (required by /usr/lib/libreadline.so.8)
awk: /home/user/Downloads/PrismLauncher-portable/lib/libc.so.6: version `GLIBC_2.38' not found (required by /usr/lib/libreadline.so.8)
awk: /home/user/Downloads/PrismLauncher-portable/lib/libc.so.6: version `GLIBC_ABI_DT_RELR' not found (required by /usr/lib/libmpfr.so.6)
awk: /home/user/Downloads/PrismLauncher-portable/lib/libc.so.6: version `GLIBC_ABI_DT_RELR' not found (required by /usr/lib/libgmp.so.10)
awk: /home/user/Downloads/PrismLauncher-portable/lib/libc.so.6: version `GLIBC_2.38' not found (required by /usr/lib/libgmp.so.10)
awk: /home/user/Downloads/PrismLauncher-portable/lib/libc.so.6: version `GLIBC_ABI_DT_RELR' not found (required by /usr/lib/libncursesw.so.6)
No missing dependencies found.
chmod: /home/user/Downloads/PrismLauncher-portable/lib/libc.so.6: version `GLIBC_ABI_DT_RELR' not found (required by chmod)
zsh: segmentation fault (core dumped)  ./PrismLauncher
```
-# the develop portable build works

getchoo committed:
build(cmake): replace INSTALL_BUNDLE with install component

Considering this doesn't affect the build, it has never made much sense
for it to be a build option or require rereconfiguration of the project
to change

Signed-off-by: Seth Flynn <getchoo@tuta.io>

getchoo commented:
This PR should see no change in the Linux portable build since https://github.com/PrismLauncher/PrismLauncher/pull/3940 (as it shouldn't actually be touching anything to do with it lol)

Edit: MSVC CI failures are unrelated

Trial97 reviewed (approved):
at least now the Linux portable works

getchoo commented:
See https://github.com/PrismLauncher/PrismLauncher/pull/3742 for MSVC builds. Our vendored zlib detection is screwed, and that fixes it (and is based directly on this branch)

getchoo commented:
Tested on macOS 15.5, the AppImage, and MinGW/MSVC again
